test/integration/conftest.py

```diff
@@ -13,16 +14,95 @@
 ADMIN_ROLE = "Conservator Administrator"
 
 
-@pytest.fixture(scope="session")
-def using_kubernetes():
-    if shutil.which("kubectl") is None:
-        return False
-    kube_services = subprocess.getoutput(
-        "kubectl --insecure-skip-tls-verify get services -o name"
+@dataclasses.dataclass
+class TestSettings:
+    server_deployment: str = ""
+    server_inside_kubernetes: bool = False
+    conservator_url: str = ""
+    mongo_url: str = ""
+    pytest_inside_docker: bool = False
+
+
+test_settings = TestSettings()
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--server-deployment",
+        choices=["docker", "kind", "minikube"],
+        default="kind",
+        help="Type of deployment for tested conservator instance",
     )
-    if "conservator-webapp" in kube_services:
-        return True
-    return False
+
+
+def pytest_configure(config):
+    # deployment type of Conservator server comes from command-line parser
+    test_settings.server_deployment = config.option.server_deployment
+    test_settings.server_inside_kubernetes = config.option.server_deployment in (
+        "kind",
+        "minikube",
+    )
+
+    # pytest runtime context (native host or inside container) comes from environment
+    test_settings.pytest_inside_docker = (
+        os.environ.get("RUNNING_IN_CLI_TESTING_DOCKER") == "True"
+    )
+
+    # if using kubernetes, make sure kubectl commands default to using correct cluster
+    if test_settings.server_deployment == "kind":
+        kube_pods = subprocess.getoutput(
+            "kubectl --insecure-skip-tls-verify config use-context kind-kind"
+        )
+    elif test_settings.server_deployment == "minikube":
+        kube_pods = subprocess.getoutput(
+            "kubectl --insecure-skip-tls-verify config use-context minikube"
+        )
```

Comment:
Is something using the `kube_pods` setting?  I see it set here but not referred to anywhere.

Reply:
Good point, I cleaned that up